XpringKit/PayIDClient.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+
+/// A client for Xpring's PayID service.
+///
+/// Inputs to this class are always in X-Addresses
+/// - SeeAlso: https://xrpaddress.info/
+///
+/// - Warning: This class is experimental and should not be used in production applications.
+public class PayIDClient {
+  /// The remote URL for the PayID service.
+  private let remoteURL: URL
+
+  /// An authorization token to use when performing updates to the PayID service.
+  private let authorizationToken: String?
+
+  /// Returns whether this client is authorized for update requests.
+  public var isAuthorizedForUpdates: Bool {
+    return authorizationToken != nil
+  }
+
+  /// Initialize a new unauthorized PayIDClient.
+  ///
+  /// - Parameters:
+  ///   - remoteURL The remote URL of the service this class will communicate with.
+  public convenience init(remoteURL: URL) {
+    self.init(remoteURL: remoteURL, authorizationToken: nil)
+  }
+
+  /// Initialize a new unauthorized PayIDClient.
+  ///
+  /// - Parameters:
+  ///   - remoteURL The remote URL of the service this class will communicate with.
+  ///   - authorizationToken An optional token to authorize write requests. Defaults to undefined.
+  public init(remoteURL: URL, authorizationToken: String? = nil) {
+    self.remoteURL = remoteURL
+    self.authorizationToken = authorizationToken
+  }
+
+  /// Retrieve the XRP Address authorized with a PayID.
+  ///
+  /// - Note: The returned value will always be in an X-Address format.
+  ///
+  /// - Parameter payID The payID to resolve for an address.
+  /// - Returns: A resutl of the operation.
+  public func address(for payID: String) -> Result<String, PayIDClientError> {
+    // TODO(keefertaylor): Implement reads from backend service.
+    return .failure(.unimplemented)
+  }
+
+  /// Update a PayID to XRP address mapping.
+  ///
+  /// - Note: The input address to this method must be in X-Address format.
+  ///
+  /// - Parameters:
+  ///   - payID The payID to update.
+  ///   - xrpAddress The new XRP address to associate with the payID.
+  /// - Returns: A boolean indicating success of the operation.
+  public func update(payID: String, to xrpAddress: String) -> Result<Bool, PayIDClientError> {
+    // TODO(keefertaylor): Implement writes from backend service.
+    return .failure(.unimplemented)
+  }
```

Comment:
I think this is fine, but keep in mind that a single payment pointer can resolve to multiple addresses.

For example, you could have a Mainnet, Testnet, and Devnet XRP address, along with a BTC address all associated with a single payment pointer.

Reply:
This is a great point for us to discuss.

1: only return one address per url per content-type?

example:
GET: doug.purdy.im CT: application/xrp -> mainnet xaddress.
GET: test.doug.purdy.im CT: application/xrp -> testnet xaddress.
GET: dev.doug.purdy.im CT: application/xrp -> devnet xaddress.

2: return N addresses per url per content-type?

GET: doug.purdy.im CT: application/xrp -> mainnet | testnet | devnet xaddresses

in any case, we need to support at else (3) three content types per PayID URL in March?

application/xrp+json
application/spsp+json
text/html (think is is @nhartner idea that I like @sappenin)

Reply:
My understanding here is that for PayID 1.0 (March 25 timeframe) we *only* support XRP. 

In the future, we'll support multiple currency types, and that would probably necessitate an upgrade of the SDK. My strawman proposal would be that the PayID server would deprecate the endpoint that serves XRP only addresses, and implement a new endpoint which deals with multi currency. The SDK would then deprecate this API which now connects to a deprecated endpoint and implement a new API / endpoint. 

My mental model was closer to Doug's (1). There's a testnet payID service at test.payid.xpring.io which returns testnet XRP addresses. There's payid.xpring.io which returns only mainnet XRP addresses.

----

@dpurdy I'm not sure what you mean:
application/xrp+json: This returns an XRP address
application/spsp+json: I don't know what this means? You're returning an ILP Payment pointer from a PayID? 
text/html: Similarly not sure. 

I think this API should give you either:
1) a string encoded address
2) a tuple of (a) string encoded address and (b) an enum saying what type of address it is (BTC, XRP, USD)
3) an array of the tuples in (2)

Based on my understanding of requirements for March, I went with 1. LMK if I'm misunderstanding.

Reply:
application/spsp4+json returns an ILP address and shared secret, not another payment pointer. A PayID that passes in spsp4 is an ILP payment pointer. 

Yes, we've acknowledged we need to specify network alongside currency. Current thinking is that we include currency in one header and network in another.